src/internal/m365/exchange/contacts_restore.go

```diff
@@ -73,8 +94,20 @@ func (h contactRestoreHandler) restore(
 	}
 
 	ctx = clues.Add(ctx, "item_id", ptr.Val(contact.GetId()))
+	collisionKey := api.ContactCollisionKey(contact)
+
+	if _, ok := collisionKeyToItemID[collisionKey]; ok {
+		log := logger.Ctx(ctx).With("collision_key", clues.Hide(collisionKey))
+		log.Debug("item collision")
```

Comment:
Just in case this was not intentional

```suggestion
		log := logger.Ctx(ctx).With("collision_key", clues.Hide(collisionKey)).Debug("item collision")
```

Reply:
Ah, `Debug()` doesn't return a logger instance, so that wouldn't work.  It'll change in the next PR or two, anyway.